packages/lib-classifier/src/store/TranscriptionReductions/TranscriptionReductions.js

```diff
@@ -0,0 +1,125 @@
+import asyncStates from '@zooniverse/async-states'
+import { flow, getEnv, types } from 'mobx-state-tree'
+
+const CONSENSUS_SCORE_TO_RETIRE = 3
+const MINIMUM_VIEW_TO_RETIRE = 5
+const REDUCER_KEY = 'alice'
+
+const TranscriptionLine = types.model({
+  consensusReached: types.boolean,
+  frame: types.number,
+  hasCollaborated: types.boolean,
+  points: types.array(types.frozen({})),
+  previousAnnotation: types.boolean,
+  textOptions: types.array(types.string),
+})
+
+const TranscriptionReductions = types
+  .model('TranscriptionReductions', {
+    caesarReducerKey: types.frozen(REDUCER_KEY),
+    error: types.maybeNull(types.frozen({})),
+    frame: types.optional(types.number, 0),
+    loadingState: types.optional(types.enumeration('state', asyncStates.values), asyncStates.initialized),
+    reductions: types.array(types.frozen({})),
+    subjectId: types.string,
+    workflowId: types.string
+  })
+
+  .views(self => {
+    function constructCoordinates (line) {
+      if (line && line.clusters_x && line.clusters_y) {
+        return line.clusters_x.map((point, i) => ({ x: line.clusters_x[i], y: line.clusters_y[i] }))
+      }
+      return []
+    }
+
+    function constructText (line) {
+      const sentences = []
+      if (line && line.clusters_text) {
+        line.clusters_text.forEach(value => {
+          value.forEach((word, i) => {
+            if (!sentences[i]) {
+              sentences[i] = []
+            }
+            if (word && word.length) {
+              sentences[i].push(word)
+            }
+          })
+        })
+      }
+      return sentences.map(value => value.join(' '));
+    }
+
+    function constructLine (annotation, options) {
+      const { minimumViews, threshold } = options
+      const { frame } = self
+      const points = constructCoordinates(annotation)
+      const textOptions = constructText(annotation)
+      return TranscriptionLine.create({
```

Comment:
I'm a bit uncertain about this only being available via a view function. The flow here is that if the transcribed reduced line does not have a high enough score, then its points and transcribed lines (from the original `clusters_text`) is available to seed new annotations for volunteers. Why not just store these on the `TranscriptionReductions` model? They could be in a map with the frame index as the identifier.

Reply:
Isn’t a map identified by index an array?

Reply:
Yes, but I'm speaking of the MST types and reusing the same model we use with identifiers and maps.

Reply:
Going back to the original question, the current consensus is a view derived from the reduced lines, which are stored on the model as `reductions`. The first version of this component had it as a separate array, stored on the model, but then I had to remember to update that array every time I changed the view eg. by changing frames.